File: libs/IXWebSocket/test/compatibility/node/devnull_client.js
//
// With ws@7.3.1
// and 
// node --version
// v13.11.0
//
// In a different terminal, start a push server:
// $ ws push_server -q
//
// $ node devnull_client.js
// messages received per second: 16643
// messages received per second: 28065
// messages received per second: 28432
// messages received per second: 22207
// messages received per second: 28805
// messages received per second: 28694
// messages received per second: 28180
// messages received per second: 28601
// messages received per second: 28698
// messages received per second: 28931
// messages received per second: 27975
//
const WebSocket = require('ws');

const ws = new WebSocket('ws://localhost:8008');

ws.on('open', function open() {
  ws.send('hello from node');
});

var receivedMessages = 0;

setInterval(function timeout() {
  console.log(`messages received per second: ${receivedMessages}`)
  receivedMessages = 0;
}, 1000);

ws.on('message', function incoming(data) {
  receivedMessages += 1;
});

File: libs/IXWebSocket/test/compatibility/ruby/devnull_client.rb
#
# $ ruby --version
# ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
# 
# Install a gem locally by setting GEM_HOME
# https://stackoverflow.com/questions/486995/ruby-equivalent-of-virtualenv
# export GEM_HOME=$HOME/local/gems
# bundle install faye-websocket
#
# In a different terminal, start a push server:
# $ ws push_server -q
#
# $ ruby devnull_client.rb
# [:open]
# Connected to server
# messages received per second: 115926
# messages received per second: 119156
# messages received per second: 119156
# messages received per second: 119157
# messages received per second: 119156
# messages received per second: 119156
# messages received per second: 119157
# messages received per second: 119156
# messages received per second: 119156
# messages received per second: 119157
# messages received per second: 119156
# messages received per second: 119157
# messages received per second: 119156
# ^C[:close, 1006, ""]
# 
require 'faye/websocket'
require 'eventmachine'

EM.run {
  ws = Faye::WebSocket::Client.new('ws://127.0.0.1:8008')

  counter = 0

  EM.add_periodic_timer(1) do
    print "messages received per second: #{counter}\n"
    counter = 0  # reset counter
  end

  ws.on :open do |event|
    p [:open]
    print "Connected to server\n"
  end

  ws.on :message do |event|
    # Uncomment the next line to validate that we receive something correct
    # p [:message, event.data]
    counter += 1
  end

  ws.on :close do |event|
    p [:close, event.code, event.reason]
    ws = nil
  end
}
